vendor/github.com/xtaci/smux/mux.go

```diff
@@ -24,15 +24,20 @@ type Config struct {
 	// MaxReceiveBuffer is used to control the maximum
 	// number of data in the buffer pool
 	MaxReceiveBuffer int
+
+	// WriteTimeout defines the default amount of time to wait before giving up
+	// on a write.
+	WriteTimeout time.Duration
```

Comment:
Added a WriteTimeout, so that all session operations will have some sane default timeout